src/Streamly/Internal/Control/ForkLifted.hs

```diff
@@ -32,11 +33,23 @@ doFork :: MonadRunInIO m
     -> RunInIO m
     -> (SomeException -> IO ())
     -> m ThreadId
-doFork action (RunInIO mrun) exHandler =
+doFork = doFork' False
+
+-- | Similar to 'doFork', but has a "bound" boolean parameter for specifying
+-- whether 'forkOS' should be used instead of 'rawForkIO'.
+{-# INLINE doFork' #-}
+doFork' :: MonadRunInIO m
+    => Bool
```

Comment:
This is like some type of configuration for `doFork`
Maybe rename this to `doForkWith` where `Bool` is the config.

Reply:
Sure; resolved in separate PR: https://github.com/composewell/streamly/pull/2406